### PR TITLE
Feature/OIDC appauth

### PR DESCRIPTION
**Pull request by jack-brinkman**

### CODE REVIEW

File: BioCollect/src/settings/GASettingsConstant.h [hunk 1]
```diff
@@ -39,4 +42,13 @@
...
 #define HUB_VIEW @"hubview"
 
+// OpenID Connect Configuration
+#define COGNITO_ENABLED false
+#define COGNITO_REGION @"ap-southeast-2"
+#define COGNITO_USER_POOL @"OOXU9GW39"
+
+// #define CLIENT_ID @"3evrluqa40pn11u2f2hqah813r" COGNITO
+#define CLIENT_ID @"oidc-mobile-auth"
+#define SCOPE @"email openid profile ala/attrs ala/roles"
```
Comment (temi) [inline BioCollect/src/settings/GASettingsConstant.h]: Looks like some are cognito scopes. will that be a problem?

File: BioCollect/src/GALogin.m [hunk 2]
```diff
@@ -60,58 +92,60 @@ - (IBAction)onClickRegister:(id)sender{
...
-        [textField resignFirstResponder];
-        [self authenticate];
+    // Retrieve the OpenID configuration
+    OIDServiceConfiguration *configuration = [GASettings getOpenIDConfig];
+    NSString *bundleId = [[[NSBundle mainBundle] bundleIdentifier] stringByReplacingOccurrencesOfString:@".testing" withString:@""];
+    NSArray *bundleParts = [bundleId componentsSeparatedByString:@"."];
+    NSString *bundleName = [bundleParts lastObject];
+    
+    // Fix bundleName for bilbyblitz app
+    if ([bundleName isEqualToString:@"tracker"]) {
```
Comment (temi) [inline BioCollect/src/GALogin.m:102]: curious to know why bundle name is changed.

Comment (jack-brinkman) [reply BioCollect/src/GALogin.m:102]: The app's bundle name on Android is `bilbyblitz`, so I added this for consistency (& so we don't need an extra redirect URL in the CAS/Cognito settings)

File: BioCollect/src/settings/GASettingsConstant.h (hunk 1 above)
Comment (jack-brinkman) [reply BioCollect/src/settings/GASettingsConstant.h]: It hasn't presented any issues during testing, however, I'll update them.

Comment (jack-brinkman) [reply BioCollect/src/settings/GASettingsConstant.h]: Updated 👍

**Review by temi — APPROVED** [review]

**Pull request merged**